### PR TITLE
Create waves

### LINKED ISSUE #128: Create waves
**Description**

Waves are based on time, the waves are infinite

### PR DESCRIPTION
**Pull request by Gabriel-Sous-a**
### Description

Added the creation of waves.

### Image/Video

N/A

### Related Issue

Closes #128 

### Does this PR introduce a breaking change? ⚠️

- [x] No
- [ ] Yes 

### Tests

- [ ] I tested all game E2E;
- [x] I just tested some scenes;
- [ ] Not applicable.

### Check list

Before submitting this pull request, please ensure the following are completed:

- [x] [My commits follow the project conventions](https://github.com/FlamingoFiestaStudio/OlympusGoneWild/wiki/ProjectBestPractices#commit-message-convections).
- [x] My code follows the project's coding style and conventions.
- [ ] I have updated the project documentation (if necessary).
- [x] My code makes sense with the design.

## Additional Notes

 The values in this PR are only temporary since the game will be fully tested in order to refine this values.

---

Thank you for your contribution to Olympus Gone Wild! Your effort is greatly appreciated, and it helps make this project better for everyone. 😄🎮🚀

### CODE REVIEW

File: main/level_1.gd [hunk 1]
```diff
@@ -8,6 +8,8 @@ extends Node2D
 @onready var enemy_scene: PackedScene = preload("res://enemies/Enemy.tscn")
 @onready var tile_set: TileSet
 @onready var player = get_node("Player")
+@onready var enemy_spawner_timer : Timer = get_node("EnemySpawnerTimer")
+@onready var enemy_container : Node2D = get_node("EnemyContainer")
```
Comment (201flaviosilva) [inline main/level_1.gd:12]:
```suggestion
@onready var enemy_spawner_timer: Timer = get_node("EnemySpawnerTimer")
@onready var enemy_container: Node2D = get_node("EnemyContainer")
```

File: main/level_1.gd [hunk 2]
```diff
@@ -42,11 +44,19 @@ func _on_enemy_spawner_timer_timeout() -> void:
...
 	new_enemy.call_deferred("initialize_enemy", player)
 
 func _on_player_shoot(Bullet: PackedScene, _position: Vector2, _direction: Vector2) -> void:
 	var new_bullet = Bullet.instantiate()
 	add_child(new_bullet)
 	new_bullet.call_deferred("start", _position, _direction)
+
+func _on_waves_timer_timeout() -> void:
+	for child in enemy_container.get_children():
+		if child.has_method("queue_free"):
```
Comment (201flaviosilva) [inline main/level_1.gd:58]: I don't think you need this verification, I think every node has that, but sure :)

File: main/level_1.gd [hunk 3]
```diff
@@ -42,11 +44,19 @@ func _on_enemy_spawner_timer_timeout() -> void:
...
+			child.queue_free()
+		
+	enemy_spawner_timer.wait_time *= 0.8
+	GameManager._set_enemy_damage(GameManager._get_enemy_damage() * 1.2)
```
Comment (201flaviosilva) [inline main/level_1.gd:62]: Add those magic number to a variable, I think in the `GameManager` could be the right place, as `const`

File: main/level_1.gd [hunk 4]
```diff
@@ -42,11 +44,19 @@ func _on_enemy_spawner_timer_timeout() -> void:
 		random_y += enemy_buffer_spawn_positive
 
 	new_enemy.position = Vector2(random_x, random_y)
-	add_child(new_enemy)
+	enemy_container.add_child(new_enemy)
```
Comment (201flaviosilva) [inline main/level_1.gd:47]: Nice work

**Review by 201flaviosilva — APPROVED** [review]
Nice Work 👌 

(Not tested in Godot, but looks good)

File: main/level_1.gd (hunk 3 above)
Comment (Gabriel-Sous-a) [reply main/level_1.gd]: i placed it as a `var` in case we want to change that throughout the game but can change top `const` if needed

**Review by Goncalo-FP — APPROVED** [review]

**Pull request merged**